### PR TITLE
Remove warnings in gr-blocks and gr-network.

### PR DESCRIPTION
**Pull request by drmpeg**

### CODE REVIEW

File: gr-network/lib/tcp_sink_impl.cc [hunk 1]
```diff
@@ -219,9 +219,6 @@ bool tcp_sink_impl::stop()
 
 void tcp_sink_impl::check_for_disconnect()
 {
-    char buff[1];
```
Comment (mbr0wn) [inline gr-network/lib/tcp_sink_impl.cc:222]: Is this correct? Is the `receive()` call maybe necessary to update `ec`?

Also, when is one supposed to call this method, and do we need it? It never gets called, and has no documented contract.

Comment (drmpeg) [reply gr-network/lib/tcp_sink_impl.cc:222]: Thanks for catching this. Also, the receive() call blocks, so probably not good in general. Seems like the function should be deleted, but maybe @ghostop14 can comment.

Comment (michaelld) [reply gr-network/lib/tcp_sink_impl.cc:222]: Looks like the call to `receive()` is meant to check whether the socket is still open (at all) which will then set the error code `ec`: https://www.boost.org/doc/libs/1_72_0/doc/html/boost_asio/reference/basic_stream_socket/receive/overload3.html . So ... seems like removing this code is a bad idea ...

**Comment by ghostop14** [comment]
Hey folks, yeah the receive has to stay to detect if the socket has been closed or not.

File: gr-network/lib/tcp_sink_impl.cc (hunk 1 above)
Comment (drmpeg) [reply gr-network/lib/tcp_sink_impl.cc:222]: I've put the code back in. But the question still remains as to whether the entire check_for_disconnect() method should be deleted.

**Comment by drmpeg** [comment]
@ghostop14 What is check_for_disconnect() for if it never gets called?

**Comment by ghostop14** [comment]
Ah, yeah.  Looks like I moved that functionality directly to the work function and checked the ec return values there rather than calling the function.  You can remove check_for_disconnect().

**Pull request merged**